pkg/scheduler/framework/plugins/clusteraffinity/azure/compute.go

```diff
@@ -0,0 +1,131 @@
+//
+//Copyright (c) Microsoft Corporation.
+//Licensed under the MIT license.
+
+// Package azure provides Azure-specific utilities and services.
+package azure
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/klog/v2"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	computev1 "go.goms.io/fleet/apis/protos/azure/compute/v1"
+	"go.goms.io/fleet/pkg/clients/azure/compute"
+	"go.goms.io/fleet/pkg/propertyprovider/azure"
+)
+
+const (
+	// AzureLocationLabelKey is the label key for Azure location
+	AzureLocationLabelKey = "fleet.azure.com/location"
+
+	// AzureSubscriptionIDLabelKey is the label key for Azure subscription ID
+	AzureSubscriptionIDLabelKey = "fleet.azure.com/subscription-id"
+)
+
+// ComputeService is the  implementation of ComputeService.
+type ComputeService struct {
```

Comment:
```suggestion
type PropertyChecker struct {
```

It can call any azure service or cache. They're all internal implementation.